Controller/Notify_Handler.php

```diff
@@ -0,0 +1,168 @@
+
+use \Psr\Log\LoggerInterface;
+use \InvalidArgumentException;
+
+/**
+ * Notify handler class.
+ *
+ * Handle Payment SDK notifications.
+ */
+class Notify_Handler extends FrontendController
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $_oLogger;
+
+    /**
+     * NotifyHandler constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->_oLogger = Registry::getLogger();
+    }
+
+    /**
+     * Request handling function.
+     *
+     * @return void
+     */
+    public function handleRequest()
+    {
+        $sPaymentName = Registry::getRequest()->getRequestParameter('pmt');
+        $oPaymentMethod = Payment_Method_Factory::create($sPaymentName);
+        $oConfig = $oPaymentMethod->getConfig();
+        $sPostData = file_get_contents('php://input');
+
+        try {
+            $oService = new TransactionService($oConfig, $this->_oLogger);
+            $oNotificationResponse = $oService->handleNotification($sPostData);
+        } catch (InvalidArgumentException $exception) {
+            $this->_oLogger->error($exception->getMessage());
+            return;
+        } catch (MalformedResponseException $exception) {
+            $this->_oLogger->error($exception->getMessage());
+            return;
+        }
+
+        // Return the response or log errors if any happen.
+        if ($oNotificationResponse instanceof SuccessResponse && $oNotificationResponse->isValidSignature()) {
+            $this->_onNotificationSuccess($oNotificationResponse);
+        } else {
+            $this->_onNotificationError($oNotificationResponse);
+        }
+    }
+
+    /**
+     * Handles success notifications
+     *
+     * @param Response $oResponse
+     *
+     * @return void
+     */
+    private function _onNotificationSuccess(Response $oResponse)
+    {
+        $this->_oLogger->debug('Notification response: ' . $oResponse->getRawData());
+        $aData = $oResponse->getData();
+
+        $sParentTransactionId = $aData['parent-transaction-id'];
+        $oTransaction = oxNew(Transaction::class);
+        if (!$oTransaction->loadWithTransactionId($sParentTransactionId)) {
+            $this->_oLogger->error('TransactionId not valid: ' . $sParentTransactionId);
+            return;
+        }
+
+        $sOrderId = $aData['order-number'];
+        $oOrder = oxNew(Oxid_Order::class);
+        if (!$oOrder->load($sOrderId)) {
+            $this->_oLogger->error('OrderId not valid: ' . $sOrderId);
+            return;
+        }
+
+        if ($oOrder->oxorder__wdoxidee_final->value) {
+            $this->_oLogger->warning('Corresponding order is already finished, nothing updated! OrderId: ' . $sOrderId);
+            return;
+        }
+
+        $oTransaction->wdoxidee_ordertransactions__ordernumber = new Field($oOrder->oxorder_oxordernr->value);
+        $oTransaction->wdoxidee_ordertransactions__transactionid = new Field($aData['transaction-id']);
```

Comment:
Please move this to separate methods as updateOrder and updateTransaction

Reply:
will be done as soon as the next payment method is implemented